hazelcast/src/test/java/com/hazelcast/config/alto/AltoConfigTest.java

```diff
@@ -46,10 +47,27 @@ public void testEventloopCount() {
     }
 
     @Test
-    public void testConfigBounds() {
+    public void testConfigValidation() {
         AltoConfig altoConfig = config.getAltoConfig();
-        assertThrows(InvalidConfigurationException.class, () -> altoConfig.setEventloopCount(0));
-        assertThrows(InvalidConfigurationException.class, () -> altoConfig.setEventloopCount(256 + 1));
+        assertThrows(IllegalArgumentException.class, () -> altoConfig.setEventloopCount(0));
+    }
+
+    @Test
+    public void testSystemProperties() {
+        System.setProperty(ALTO_ENABLED.getName(), "true");
+        System.setProperty(ALTO_EVENTLOOP_COUNT.getName(), "3");
+        HazelcastInstance hz = createHazelcastInstance(config);
+        assertTrue(isTpcEnabled(hz));
+        assertEquals(3, getEventloopCount(hz));
+    }
+
+    @Test
+    public void testConfigProperties() {
+        config.setProperty(ALTO_ENABLED.getName(), "true");
+        config.setProperty(ALTO_EVENTLOOP_COUNT.getName(), "3");
+        HazelcastInstance hz = createHazelcastInstance(config);
+        assertTrue(isTpcEnabled(hz));
+        assertEquals(3, getEventloopCount(hz));
```

Comment:
Minor: maybe add a test that verifies the system properties overrides configuration, when both are present

Reply:
https://github.com/hazelcast/hazelcast/pull/23319/commits/86691d6ffc6b40f987ad9926a50a7553c4a08f3f